drivers/power/adp1050/iio_adp1050.c

```diff
@@ -0,0 +1,878 @@
+		.show = adp1050_iio_read_vout,
+		.store = adp1050_iio_write_vout,
+		.priv = ADP1050_VOUT_OFFSET_VALUE
+	},
+	{
+		.name = "freq_sync",
+		.show = adp1050_iio_read_enable,
+		.store = adp1050_iio_write_enable,
+		.priv = ADP1050_FREQ_SYNC_ENABLE,
+		.shared = IIO_SHARED_BY_ALL
+	},
+	{
+		.name = "freq_sync_available",
+		.show = adp1050_iio_read_enable_available,
+		.priv = ADP1050_FREQ_SYNC_ENABLE,
+		.shared = IIO_SHARED_BY_ALL
+	},
+	{
+		.name = "feedforward",
+		.show = adp1050_iio_read_enable,
+		.store = adp1050_iio_write_enable,
+		.priv = ADP1050_FEEDFORWARD_ENABLE,
+		.shared = IIO_SHARED_BY_ALL
+	},
+	{
+		.name = "feedforward_available",
+		.show = adp1050_iio_read_enable_available,
+		.priv = ADP1050_FEEDFORWARD_ENABLE,
+		.shared = IIO_SHARED_BY_ALL
+	}
+};
+
+static struct iio_attribute adp1050_debug_attrs[] = {
+	{
+		.name = "status_vout",
+		.show = adp1050_iio_read_status,
+		.priv = ADP1050_STATUS_VOUT_TYPE
+	},
+	{
+		.name = "status_input",
+		.show = adp1050_iio_read_status,
+		.priv = ADP1050_STATUS_INPUT_TYPE,
+	},
+	{
+		.name = "status_temperature",
+		.show = adp1050_iio_read_status,
+		.priv = ADP1050_STATUS_TEMPERATURE_TYPE,
+	},
+	{
+		.name = "status_cml",
+		.show = adp1050_iio_read_status,
+		.priv = ADP1050_STATUS_CML_TYPE
+	},
+	{
+		.name = "status_word",
+		.show = adp1050_iio_read_status,
+		.priv = ADP1050_STATUS_WORD_TYPE
+	}
+};
+
+static struct iio_channel adp1050_channels[] = {
+	{
+		.name = "vin",
+		.ch_type = IIO_VOLTAGE,
+		.indexed = 0,
+		.channel = 0,
+		.address = 0,
+		.attributes = adp1050_input_attrs,
+		.ch_out = false
+	},
+	{
+		.name = "iin",
+		.ch_type = IIO_CURRENT,
+		.indexed = 0,
+		.channel = 1,
+		.address = 1,
+		.attributes = adp1050_input_attrs,
+		.ch_out = false,
+	},
+	{
+		.name = "vout",
+		.ch_type = IIO_VOLTAGE,
+		.indexed = 0,
+		.channel = 2,
+		.address = 2,
+		.attributes = adp1050_input_attrs,
+		.ch_out = false,
+	},
+	{
+		.name = "temperature",
+		.ch_type = IIO_TEMP,
+		.indexed = 0,
+		.channel = 3,
+		.address = 3,
+		.attributes = adp1050_input_attrs,
+		.ch_out = false,
+	},
+	{
+		.name = "outa",
+		.ch_type = IIO_VOLTAGE,
+		.indexed = 0,
+		.channel = 4,
+		.address = 4,
+		.attributes = adp1050_output_attrs,
+		.ch_out = false,
+	},
+	{
+		.name = "outb",
+		.ch_type = IIO_VOLTAGE,
+		.indexed = 0,
+		.channel = 5,
+		.address = 5,
+		.attributes = adp1050_output_attrs,
+		.ch_out = true,
+	},
+	{
+		.name = "sr1",
+		.ch_type = IIO_VOLTAGE,
+		.indexed = 0,
+		.channel = 6,
+		.address = 6,
+		.attributes = adp1050_output_attrs,
+		.ch_out = true,
+	},
+	{
+		.name = "sr2",
+		.ch_type = IIO_VOLTAGE,
+		.indexed = 0,
+		.channel = 7,
+		.address = 7,
+		.attributes = adp1050_output_attrs,
+		.ch_out = true,
+	}
+};
+
+static struct iio_device adp1050_iio_dev = {
+	.debug_reg_read = (int32_t (*)())adp1050_iio_reg_read,
+	.debug_reg_write = (int32_t (*)())adp1050_iio_reg_write,
+	.pre_enable = (int32_t (*)())adp1050_iio_update_channels,
```

Comment:
pre_enable function is used only with buffer readings. and you do not have read_dev defined so in this case it is not used. either remove it or add the buffer readings support if you intended to

Reply:
Pre_enable function will be removed.